### PR TITLE
Boss bar fix 

### PR DESCRIPTION
**Pull request by Moistcrafter**
## Description
fixes some silly bar for some bosses, also fixes sub-names things under the boss bar naame
ty hiro for some code

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.

## Type of changes

- [x] Bug fix
- [ ] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by Moistcrafter** [comment]
Genshin Impact Screenshot 2023 05 27 - 12 46 22 53
Genshin Impact Screenshot 2023 05 27
![Genshin Impact Screenshot 2023 05 27 - 12 37 58 30
 - 12 44 51 46](https://github.com/Grasscutters/Grasscutter/assets/63527712/8eb329f5-22e2-4455-9877-9ddb533c0599)

File: src/main/java/emu/grasscutter/data/excels/MonsterData.java [hunk 1]
```diff
@@ -94,6 +94,18 @@ public void onLoad() {
...
+    
+        if (this.describeData == null){
+            return;
+        }       
+        for(Entry<Integer, MonsterSpecialNameData> entry: GameData.getMonsterSpecialNameDataMap().entrySet()) {
+            if (entry.getValue().getSpecialNameLabId() == this.getDescribeData().getSpecialNameLabId()){
+                this.specialNameId = entry.getKey();
+                break;
+            }
+        }
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/data/excels/MonsterData.java]:
```suggestion
        for (var entry : GameData.getMonsterSpecialNameDataMap().entrySet()) {
            if (entry.getValue().getSpecialNameLabId() == this.getDescribeData().getSpecialNameLabId()) {
                this.specialNameId = entry.getKey();
                break;
            }
        }
```

File: src/main/java/emu/grasscutter/data/excels/MonsterData.java [hunk 2]
```diff
@@ -94,6 +94,18 @@ public void onLoad() {
                 this.weaponId = id;
             }
         }
+
+        this.describeData = GameData.getMonsterDescribeDataMap().get(this.getDescribeId());
+    
+        if (this.describeData == null){
+            return;
+        }
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/data/excels/MonsterData.java]:
```suggestion
        if (this.describeData == null) return;
```

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/data/excels/MonsterData.java [hunk 3]
```diff
@@ -94,6 +94,17 @@ public void onLoad() {
                 this.weaponId = id;
             }
         }
+
+        this.describeData = GameData.getMonsterDescribeDataMap().get(this.getDescribeId());
+    
+        if (this.describeData == null) return;
+
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/data/excels/MonsterData.java]:
```suggestion
```

**Review by KingRainbow44 — APPROVED** [review]

**Pull request merged**